samples/snippets/src/main/java/aiplatform/CreateTrainingPipelineCustomJobSample.java

```diff
@@ -0,0 +1,124 @@
+        project, displayName, modelDisplayName, containerImageUri, baseOutputDirectoryPrefix);
+  }
+
+  static void createTrainingPipelineCustomJobSample(
+      String project,
+      String displayName,
+      String modelDisplayName,
+      String containerImageUri,
+      String baseOutputDirectoryPrefix)
+      throws IOException {
+    PipelineServiceSettings settings =
+        PipelineServiceSettings.newBuilder()
+            .setEndpoint("us-central1-aiplatform.googleapis.com:443")
+            .build();
+    String location = "us-central1";
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (PipelineServiceClient client = PipelineServiceClient.create(settings)) {
+      JsonObject jsonMachineSpec =
+          Json.createObjectBuilder().add("machineType", "n1-standard-4").build();
+      JsonArray jsonArgs =
+          Json.createArrayBuilder()
+              // AIP_MODEL_DIR is set by the service according to baseOutputDirectory.
+              .add("--model_dir=$(AIP_MODEL_DIR)")
+              .build();
+      JsonObject jsonContainerSpec =
+          Json.createObjectBuilder()
+              // A working docker image can be found at
+              // gs://cloud-samples-data/ai-platform/mnist_tfrecord/custom_job
+              .add("imageUri", containerImageUri)
+              .add("args", jsonArgs)
+              .build();
+      JsonObject jsonJsonWorkerPoolSpec0 =
+          Json.createObjectBuilder()
+              .add("replicaCount", 1)
+              .add("machineSpec", jsonMachineSpec)
+              .add("containerSpec", jsonContainerSpec)
+              .build();
+      JsonArray jsonWorkerPoolSpecs =
+          Json.createArrayBuilder().add(jsonJsonWorkerPoolSpec0).build();
+      JsonObject jsonBaseOutputDirectory =
+          Json.createObjectBuilder()
+              // The GCS location for outputs must be accessible by the project's AI Platform
+              // service account.
+              .add("output_uri_prefix", baseOutputDirectoryPrefix)
+              .build();
+      JsonObject jsonTrainingTaskInputs =
+          Json.createObjectBuilder()
+              .add("workerPoolSpecs", jsonWorkerPoolSpecs)
+              .add("baseOutputDirectory", jsonBaseOutputDirectory)
+              .build();
+      Value.Builder trainingTaskInputsBuilder = Value.newBuilder();
+      JsonFormat.parser().merge(jsonTrainingTaskInputs.toString(), trainingTaskInputsBuilder);
+      Value trainingTaskInputs = trainingTaskInputsBuilder.build();
+      String trainingTaskDefinition =
+          "gs://google-cloud-aiplatform/schema/trainingjob/definition/custom_task_1.0.0.yaml";
+      String imageUri = "gcr.io/cloud-aiplatform/prediction/tf-cpu.1-15:latest";
```

Comment:
:latest is typically frowned upon.

Reply:
Acknowledged. @dizcology can give more context.

See Python canonical:
https://github.com/googleapis/python-aiplatform/blob/master/samples/snippets/create_training_pipeline_custom_job_sample.py#L58